src/games/GameDetails.js

```diff
@@ -1,53 +1,52 @@
 import { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
 import { useParams } from "react-router-dom";
-import PugApi from "../api/api";
 import LoadingSpinner from "../common/LoadingSpinner";
+import { fetchGame, fetchGames } from "../store/games/gamesSlice";
 
 function GameDetails() {
-    
-  const { gameId }= useParams();
-  console.debug("GameDetail", "gameId=", gameId);
-
-  const [game, setGame] = useState(null)
-  const [details, setDetails] = useState(null)
-  const [comments, setComments] = useState(null)
-  const [players, setPlayers] = useState(null)
-
-  useEffect(()=>{
-    async function getGame(){
-      const game = await PugApi.getGame(gameId) 
-      const {details, comments, players } = game
-      setGame(game)
-      setDetails(details)
-      setComments(comments)
-      setPlayers(players)
-    }
+  const { gameId } = useParams();
 
-    getGame()
-  }, [gameId])
+  const dispatch = useDispatch();
+  const gameStatus = useSelector((state) => state.games.status);
+  const games = useSelector((state) => state.games.entities);
+  const error = useSelector((state) => state.games.errors);
 
-  console.log(gameId, game, comments, players)
-
-  if(!game) return <LoadingSpinner />
+  console.debug("GameDetail", "gameId=", gameId);
 
-  return (
-    <div>
-      <h1>Game Details: {gameId}</h1>
-      <ul>
-        <li>id: {details.id}</li>
-        <li>title: {details.title}</li>
-        <li>description: {details.description}</li>
-        <li>date: {details.date}</li>
-        <li>time: {details.time}</li>
-        <li>address: {details.address}</li>
-        <li>city: {details.city}</li>
-        <li>state: {details.state}</li>
-        <li>created on: {details.createdOn}</li>
-        <li>game host: {details.createdBy}</li>
-        <li>days until game: {details.daysDiff}</li>
-      </ul>
-    </div>
-  );
+  useEffect(() => {
+    if (gameStatus === "idle") {
+      dispatch(fetchGames());
```

Comment:
Not sure how the Game Details is intended but if you're on the route `/games/$id` i dont think you'll end up fetching all the games on this page

Reply:
updated to fetchGame(gameId) 
fetchGame retrieves game data and hydrates gameSlice, userSlice, and commentSlice